QEfficient/transformers/models/modeling_auto.py

```
@@ -1498,6 +1509,9 @@ def compile(
             decode_specialization.update({"num_logits_to_keep": num_speculative_tokens + 1}) if self.is_tlm else ...
             specializations.append(decode_specialization)
 
+        if compiler_options.pop("img_size", None):
+            logger.warning("img_size is not a valid argument for Text-to-Text Model.")
+
```

Why? out compiler will fail if user passes img_size .
Let it fail

---

Since this img_size is passed as an args to compiler which is not an acceptable argument by Causal Models, it throws error as Invalid option, so this needs to be popped out. And while popping if its value is not None then throwing a warning.

---

We can throw an error with the proper error message.